### PR TITLE
HDDS-10390. MiniOzoneCluster to support S3G

### PR DESCRIPTION
**Pull request by ivandika3**
## What changes were proposed in this pull request?

MiniOzoneCluster should support creating S3G instances and provide access to S3G via an AWS S3 client (AWS Java SDK).

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10390

## How was this patch tested?

Added some tests for AWS S3 SDK V1.

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @ivandika3 for the patch, integration LGTM.  Haven't checked actual test methods in detail.

New tests are reasonably quick:

```
[INFO] Running org.apache.hadoop.ozone.s3.awssdk.v1.TestS3SDKV1WithRatis
[INFO] Tests run: 20, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 47.155 s - in org.apache.hadoop.ozone.s3.awssdk.v1.TestS3SDKV1WithRatis
[INFO] Running org.apache.hadoop.ozone.s3.awssdk.v1.TestS3SDKV1
[INFO] Tests run: 20, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 32.808 s - in org.apache.hadoop.ozone.s3.awssdk.v1.TestS3SDKV1
```

**Comment by ivandika3** [comment]
@adoroszlai Thanks for checking this out.

Currently the AWS SDK tests are not extensive. We can raise a parent ticket to add more extensive tests (similar to https://github.com/ceph/s3-tests).

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneClusterImpl.java [hunk 1]
```diff
@@ -286,6 +312,54 @@ public OzoneClient newClient() throws IOException {
     return client;
   }
 
+  @Override
+  public AmazonS3 newS3Client() {
+    // TODO: Parameterize tests between Virtual host style and Path style
+    return createS3Client(true);
+  }
+
+  public AmazonS3 createS3Client(boolean enablePathStyle) {
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneClusterImpl.java:321]: In a follow-up task I think we should add a separate factory class for creating the client, and remove `AmazonS3` from `MiniOzoneCluster` interface.

**Review by adoroszlai — APPROVED** [review]

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneClusterImpl.java (hunk 1 above)
Comment (ivandika3) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneClusterImpl.java:321]: Good idea, raised https://issues.apache.org/jira/browse/HDDS-11576

**Pull request merged**